### PR TITLE
Upgrade to semconv/v1.17.0

### PR DESCRIPTION
**Pull request by MrAlias**
- Upgrade all `semconv` use to the new, unreleased, `v1.17.0` package
- Upgrade HTTP semantics to use the new `httpconv` and `netconv` pacakges, and update any direct `semconv` references to rely on new semantic conventions
  - Related specification changes: https://github.com/open-telemetry/opentelemetry-specification/pull/2469
- Upgrade messaging semantics to use the new semantic conventions
  - Related specification changes: https://github.com/open-telemetry/opentelemetry-specification/pull/2957

### CODE REVIEW

**Comment by MrAlias** [comment]
I'm looking into the test failure on Windows for otelgrpc...

**Comment by MrAlias** [comment]
> I'm looking into the test failure on Windows for otelgrpc...

Fixed in d18cfe22541e12c5b45b1e99fcc6ec6dc466775c. Looks like the issue was on Windows the connection setup to a fake target hung indefinitely causing a goroutine to be abandoned. This was resolved by adding the `ctxDialer` and using it to actually do the connection setup for `grpc.DialContext`.

File: instrumentation/github.com/gorilla/mux/otelmux/mux.go [hunk 1]
```diff
@@ -135,24 +136,28 @@ func (tw traceware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-	spanStatus, spanMessage := semconv.SpanStatusFromHTTPStatusCodeAndSpanKind(rrw.status, oteltrace.SpanKindServer)
-	span.SetAttributes(attrs...)
-	span.SetStatus(spanStatus, spanMessage)
+	if rrw.status > 0 {
+		span.SetAttributes(semconv.HTTPStatusCodeKey.Int(rrw.status))
+	}
+	span.SetStatus(httpconv.ServerStatus(rrw.status))
+	if rrw.status > 0 {
+		span.SetAttributes(semconv.HTTPStatusCodeKey.Int(rrw.status))
+	}
```
Comment (Aneurysm9) [inline instrumentation/github.com/gorilla/mux/otelmux/mux.go:162]:
```suggestion
```

This seems to be duplicated.

**Review by Aneurysm9 — APPROVED** [review]

**Review by MadVikingGod — APPROVED** [review]

File: instrumentation/net/http/otelhttp/handler.go [hunk 2]
```diff
@@ -211,7 +215,10 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	setAfterServeAttributes(span, bw.read, rww.written, rww.statusCode, bw.err, rww.err)
 
 	// Add metrics
-	attributes := append(labeler.Get(), semconv.HTTPServerMetricAttributesFromHTTPRequest(h.operation, r)...)
+	attributes := append(labeler.Get(), httpconv.ServerRequest(h.server, r)...)
+	if rww.statusCode > 0 {
+		attributes = append(attributes, semconv.HTTPStatusCodeKey.Int(rww.statusCode))
+	}
 	h.counters[RequestContentLength].Add(ctx, bw.read, attributes...)
```
Comment (mlaflamm) [inline instrumentation/net/http/otelhttp/handler.go:222]: `ServerRequest` returns trace attributes not appropriate for metric. This change is adding high cardinality attributes to metric (#3765).

**Pull request merged**